pallets/collective-proxy/src/lib.rs

```diff
@@ -98,19 +152,22 @@ pub mod pallet {
 		})]
         pub fn execute_call(
             origin: OriginFor<T>,
+            filter: Option<T::CallFilter>,
```

Comment:
Interesting argument choice - why not choose `T::AccountId` instead?

Reply:
It's the design choice that I had to make. As I've said in PR's description, I tried to make minimal changes into the pallet that would add multiple proxies support but would keep its current design as much as possible (because I don't know details about the use case). I tried to make assumptions on use case and imagine, how pallet would be used basing on its current design. So I decided to go this way (proxy calls based on the needed filters). But again it's only my assumption how it would be used. And obviously it's pretty straightforward to rework it into the opposite way (proxy call based on needed proxy account).

Reply:
The purpose of the pallet was to allow non-EoA accounts to act on behalf of a custom origin type.
Practically, this is used for pallet-collective origins for implementing governance.
You can check this in one of the runtimes, e.g. `shibuya-runtime`.

My comment here was more related to why choose the filter as argument, and not the account Id.
It seems very strange to specify a call filter from which the proxy account is derived - IMO it would make it cleaner if user directly specifies the account Id on whose behalf they want to execute a call. 
Based on the account Id argument, you check the call filter & decide whether call is allowed or not.
That's how `pallet-proxy` does anyways.

Reply:
Thanks for the context! I looked through runtime and the usage is more clear now. I reworked to AccountId.